CSharpSourceCode/Abilities/AbilityManagerMissionLogic.cs

```diff
@@ -201,9 +201,11 @@ public override void OnAgentCreated(Agent agent)
         {
             if (IsCastingMission(Mission))
             {
-                if (agent.IsAbilityUser())
+                var character = agent.Character as CharacterObject;
+                bool canHasArtillery = (Game.Current.GameType is Campaign) && character != null && character.IsHero && character.HeroObject.IsPartyLeader;
```

Comment:
This should be driven by a new character attribute instead of being hardcoded to partyleader heroes. We might want engineer companions or special NPCs with this ability as well. Create a "CanPlaceArtillery" character attribute string and check for that with character.HasAttribute("CanPlaceArtillery") instead.

Reply:
Ok. I've created Agent extension CanPlaceArtillery instead and moved this checking to AbilityComponent